editoast/src/views/infra/routes.rs

```diff
@@ -249,57 +280,114 @@ mod tests {
-                    "rt.DD5->DE1",
-                    "rt.DE0->buffer_stop.4",
-                ],
+                (
+                    vec![
+                        "rt.DD2->DD6",
+                        "rt.DD4->DD0",
+                        "rt.DD5->DE1",
+                        "rt.DE0->buffer_stop.4",
+                    ],
+                    vec![("PD0".to_string(), vec!["STATIC".to_string()])],
+                ),
             ),
             (
                 json!({ "PH0": "A1_B1" }), // double_slip_switch
-                vec!["rt.DG0->DG3", "rt.DG1->DD7"],
+                (
+                    vec!["rt.DG0->DG3", "rt.DG1->DD7"],
+                    vec![("PH0".to_string(), vec!["A1_B1".to_string()])],
+                ),
             ),
             (
                 json!({ "PH1": null }), // all routes crossing PH1
-                vec![
-                    "rt.DG2->DH1",
-                    "rt.DH2->DG4",
-                    "rt.DH2->buffer_stop.7",
-                    "rt.DH3->DH1",
-                ],
+                (
+                    vec![
+                        "rt.DG2->DH1",
+                        "rt.DH2->DG4",
+                        "rt.DH2->buffer_stop.7",
+                        "rt.DH3->DH1",
+                    ],
+                    vec![(
+                        "PH1".to_string(),
+                        vec!["A_B1".to_string(), "A_B2".to_string()],
+                    )],
+                ),
             ),
             (
                 json!({ "PA0": "A_B1", "PA2": "A_B1" }),
-                vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                (
+                    vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                    vec![
+                        ("PA0".to_string(), vec!["A_B1".to_string()]),
+                        ("PA2".to_string(), vec!["A_B1".to_string()]),
+                    ],
+                ),
             ),
             (
                 json!({ "PA0": "A_B1", "PA2": null }),
-                vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                (
+                    vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                    vec![
+                        ("PA0".to_string(), vec!["A_B1".to_string()]),
+                        ("PA2".to_string(), vec!["A_B1".to_string()]),
+                    ],
+                ),
             ),
         ];
 
         let app = create_test_service().await;
         let small_infra = small_infra(db_pool()).await;
 
-        for (params, mut expected) in tests {
+        fn compare_result(got: RoutesFromNodesPositions, expected: RoutesFromNodesPositions) {
+            let mut got_routes = got.routes;
+            let mut expected_routes = expected.routes;
+            got_routes.sort();
+            expected_routes.sort();
+            assert_eq!(got_routes, expected_routes);
+
+            let process_available_positions = |positions: &HashMap<String, Vec<String>>| {
```

Comment:
```suggestion
            fn process_available_positions(positions: &HashMap<String, Vec<String>>) -> Vec<String, Vec<String>> {
```

Making a closure this way usually suggests you're capturing some value, `fn` makes it clear that you're not.